include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -989,6 +986,12 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::evaluate(
       fast_path)
     {
       // fast path with tensor product evaluation
+      const auto unit_points =
+        mapping_info->get_unit_points(current_cell_index, current_face_number);
+
+      if (!is_reinitialized || n_q_points != unit_points.size())
+        reinit(numbers::invalid_unsigned_int);
```

Comment:
Can you add a comment here as to why this is necessary? Without it, a reader might wonder why this `reinit` call is inside `evaluate`.